tests/tt_metal/tt_metal/common/command_queue_fixture.hpp

```diff
@@ -0,0 +1,134 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "gtest/gtest.h"
+#include "dispatch_fixture.hpp"
+#include "hostdevcommon/common_values.hpp"
+#include "impl/device/device.hpp"
+#include "tt_cluster_descriptor_types.h"
+#include "tt_metal/host_api.hpp"
+#include "tt_metal/detail/tt_metal.hpp"
+#include "tt_metal/test_utils/env_vars.hpp"
+#include "tt_metal/impl/kernels/kernel.hpp"
+#include "tt_metal/common/tt_backend_api_types.hpp"
+#include "tt_metal/llrt/rtoptions.hpp"
+
+class CommandQueueFixture : public DispatchFixture {
+   protected:
+    tt::tt_metal::Device* device_;
+    void SetUp() override {
```

Comment:
Generally c'tor / d'tor are preferred over SetUp / TearDown.  https://google.github.io/googletest/faq.html#CtorVsSetUp

Is it required here because `CloseDevice` isn't `noexcept` (why isn't it `noexcept`? 🤔 )